### PR TITLE
bump gcp driver and sidecar versions

### PR DESCRIPTION
**Pull request by RomanBednar**
#### What type of PR is this?
/kind bug
/kind failing-test

#### What this PR does / why we need it:
Snapshot tests broke due to API version mismatch: https://github.com/kubernetes/kubernetes/pull/110021#issuecomment-1135751636

Bumping driver and sidecar version should fix the issue.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:
Did a quick sanity check with:

```
$ make WHAT=test/e2e/e2e.test && _output/bin/e2e.test --provider=gce --gce-zone=europe-west3 --ginkgo.focus=".*pd.csi.storage.gke.io.*should provision storage with snapshot data source.*" 
...
Ran 2 of 7033 Specs in 288.726 seconds
SUCCESS! -- 2 Passed | 0 Failed | 0 Pending | 7031 Skipped
PASS
```

#### Does this PR introduce a user-facing change?

```release-note

```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by mattcary** [comment]
Let's follow Xing's advice... generally it should match versions in stable-master in the gcp-compute-pd-csi-driver repo.

Also eventually all this will be torn out in favor of testing the pd csi driver through cloud-provider-gcp, so we only have one set of manifests to push.

File: test/e2e/testing-manifests/storage-csi/gce-pd/controller_ss.yaml [hunk 1]
```diff
@@ -21,7 +21,7 @@ spec:
       serviceAccountName: csi-gce-pd-controller-sa
       containers:
         - name: csi-snapshotter
-          image: k8s.gcr.io/sig-storage/csi-snapshotter:v3.0.3
+          image: k8s.gcr.io/sig-storage/csi-snapshotter:v6.0.0
```
[the remark this answers is not in the document]
Comment (RomanBednar) [reply test/e2e/testing-manifests/storage-csi/gce-pd/controller_ss.yaml]: Done.

**Pull request merged**